app/controllers/users/approvals_controller.rb

```diff
@@ -22,4 +26,15 @@ def reject
     render "users/approvals/approve"
   end
 
+  def check_user
```

Comment:
Authenticate_user checks if the user is signed in I think. Check_user checks if the signed in user matches the user being accessed.